runtime/compiler/optimizer/SPMDParallelizer.cpp

```diff
@@ -492,6 +492,14 @@ bool TR_SPMDKernelParallelizer::visitTreeTopToSIMDize(TR::TreeTop *tt, TR_SPMDKe
                }
             }
 
+         TR::ILOpCode scalarOp = node->getOpCode();
+         TR::ILOpCodes vectorOpCode = TR::ILOpCode::convertScalarToVector(scalarOp.getOpCodeValue());
+         TR::SymbolReference *symRef = node->getSymbolReference();
+         TR::SymbolReference *vecSymRef = pSPMDInfo->getVectorSymRef(symRef);
+
+         if (isCheckMode && vectorOpCode == TR::BadILOp)
```

Comment:
would it be better for this to read more like:
```if (vectorOpCode == TR::BadIlOp)
    {
    TR_ASSERT_FATAL(isCheckMode, "BadILOp encountered during SPMD parallelization which was not seen during checking mode at node n%dn!", node->getGlobalIndex());
     return false;
    }```
This would guard against a bad IL op that might pop up later

Reply:
We have an assert for this purpose few lines down. Maybe we should make it FATAL. But please note that this was not the problem in this case. In this case, we skipped the check all together for nodes whose RHS is loop invariant.

Reply:
I see - if there is an assert later on that is fine - I was just trying to make sure the BadILOp was caught with an assert so I'm happy to yield to the judgement of @gita-omr on whether this is needed. Making the assert fatal may help ensure we get compile crashes at the time of generating a bad opcode rather than falling over later in codegen trying to evaluate it.